[BUGFIX] Prevent softlock caused by removing all keybinds for a UI control

VioletSnowLeopard opened the pull request:
<!-- Please read the Contributing Guide (https://github.com/FunkinCrew/Funkin/blob/main/docs/CONTRIBUTING.md) before submitting this PR. -->
## Does this PR close any issues? If so, link them below.
#3205
## Briefly describe the issue(s) fixed.
Currently, the game allows you to remove all keybinds for any control. This includes controls important for navigating menus, like DOWN, UP, ACCEPT, and BACK. If you remove all keybinds for DOWN, then move up, you cannot reset DOWN. This PR prevents removing a keybind for these important controls if doing so would result it that key having no keybinds.

(The issue linked above talks about setting ENTER as a BACK key. I tested this, and ENTER still works as ACCEPT for the "Press ENTER to start the game" while bound to BACK)

I wouldn't recommend testing breaking your controls unless you know how to copy and replace your savefile.

NOTE: When you rebind a UI control, you cannot use that new keybind until you leave the controls menu. This is because the game does not load the new control scheme you made until after the controls menu is exited. This doesn't apply when you remove a keybind for a UI control, however. When you remove a keybind for a UI control, you won't be able to use it in the controls menu without having to leave it first.

## Include any relevant screenshots or videos.
This video shows me breaking the DOWN key. After I remove both keybinds for it and move up, I cannot go back to fix it.

https://github.com/user-attachments/assets/3ca4547f-2eb4-485c-860d-de984a4b2ef4

This video shows what my change does. I remove one keybind for DOWN, but the game won't let me remove the other one.

https://github.com/user-attachments/assets/b956f67b-12ac-4285-9198-d2fc472cff9c

This video shows me being able to use ENTER to start the game even though I keybound it to BACK

https://github.com/user-attachments/assets/9d479307-c32e-4817-9fa8-f635c557bfc5

AbnormalPoof reviewed (commented):
Would it be possible to have a popup display that says something like "You can't unbind this key!" to let the user know instead of silently not allowing the key to be unbound?

VioletSnowLeopard commented:
I added a popup to alert the user that they cannot remove the keybind:


https://github.com/user-attachments/assets/f72fee2a-2e6e-4775-9e3a-7b67931fede9

VioletSnowLeopard commented:
Just fixed a weird bug I found in the controls menu. When you leave the menu with one key bound to a UI control, then return, the control will appear duplicated. This duplicate doesn't actually exist, as you can see by me replacing one with "S". When I leave and go back, they both become "S".

https://github.com/user-attachments/assets/7470d522-8459-4e26-8304-1e93a20c69d5

Turns out this happens because the `getInputsFor` function inside the `getInput` function inside the `InputItem` class returns a list that duplicates the values three times sometimes:

![Screenshot 2025-03-29 131724](https://github.com/user-attachments/assets/a45858c5-2e01-4ea0-a479-8000497f7861)

I think this only happens for controls that had keybinds changed as this did not happen for the RESET control. This bug causes the length of a list that has one value to have three, which won't allow the program to figure out that there is only one distinct key bound to that control.

I added `list = list.distinct()` after this list is generated to ensure that the list only contains unique values to fix this issue.

VioletSnowLeopard commented:
This bug that I found and fixed may be related to #2780.
